src/lib/assignment.ts

```diff
@@ -21,6 +21,43 @@ type Yaml = {
   section: string[][]
 }
 
+export type Penalty = {
+  id: number
+  datetime: Date
+  percent: number
+  message: string
+}
+
+type PenaltyRaw = {
+  id: number
+  datetime: string
+  percent: number
+  message: string
+}
+
+export type AssignmentSettings = {
+  enableResetAssignmentByStudent?: boolean
+  disableDownloadByStudent?: boolean
+  visibilityOnDisabled?: string, // "READ_ONLY", "NO_ACCESS",
```

Comment:
probably could be string literal type (https://www.typescriptlang.org/docs/handbook/release-notes/typescript-1-8.html#string-literal-types), like
`visibilityOnDisabled?: 'READ_ONLY' | 'NO_ACCESS'
the same for other includes

Reply:
was thinking about it, but didn't want to introduce a new version of the library with a string parameter value change.